stdlib/messaging/kafka/src/main/ballerina/src/kafka/Module.md

```diff
@@ -0,0 +1,146 @@
+
+service kafkaService on consumer {
+
+    resource function onMessage(kafka:ConsumerAction consumerAction, kafka:ConsumerRecord[] records) {
+        // Dispatched set of Kafka records to service, We process each one by one.
+        foreach var kafkaRecord in records {
+            processKafkaRecord(kafkaRecord);
+        }
+    }
+}
+
+function processKafkaRecord(kafka:ConsumerRecord kafkaRecord) {
+    byte[] serializedMsg = kafkaRecord.value;
+    string msg = encoding:byteArrayToString(serializedMsg);
+    // Print the retrieved Kafka record.
+    io:println("Topic: " + kafkaRecord.topic + " Received Message: " + msg);
+}
+````
+
+Please find the consumer parameters below:
+
+| Parameter | Description  |
+| :---   | :- |
+| bootstrapServers | The list of host and port pairs, which are the addresses of the Kafka brokers in a "bootstrap" Kafka cluster. |
+| groupId | The Identifier of the consumer group. |
+| topics | The topics that must be listened by this consumer. |
+| pollingInterval | The time interval that a consumer polls the topic. |
+| offsetReset | Offset reset strategy if no initial offset. |
+| partitionAssignmentStrategy | Strategy class for handling the partition assignment among consumers. |
+| metricsRecordingLevel | Metrics recording level. |
+| metricsReporterClasses | Metrics reporter classes. |
+| clientId | Identifier to be used for server side logging. |
+| interceptorClasses | Interceptor classes to be used before sending records. |
+| isolationLevel | Transactional message reading method. Use "read_committed" to read committed messages only in transactional mode when poll() is called. Use "read_uncommitted" to read all the messages, even the aborted ones. |
+| properties | Additional properties if required. |
+| sessionTimeout | Timeout used to detect consumer failures when heartbeat threshold is reached. |
+| heartBeatInterval | Expected time between heartbeats. |
+| metadataMaxAge | Maximum time to force a refresh of metadata. |
+| autoCommitInterval | Auto committing interval for commit offset, when auto-commit is enabled. |
+| maxPartitionFetchBytes | The maximum amount of data per-partition the server returns. |
+| sendBuffer | Size of the TCP send buffer (SO_SNDBUF). |
+| receiveBuffer | Size of the TCP receive buffer (SO_RCVBUF). |
+| fetchMinBytes | Minimum amount of data the server should return for a fetch request. |
+| fetchMaxBytes | Maximum amount of data the server should return for a fetch request. |
+| fetchMaxWait | Maximum amount of time the server will block before answering the fetch request. |
+| reconnectBackoffMax | Maximum amount of time in milliseconds to wait when reconnecting. |
+| retryBackoff | Time to wait before attempting to retry a failed request. |
+| metricsSampleWindow | Window of time a metrics sample is computed over. |
+| metricsNumSamples | Number of samples maintained to compute metrics. |
+| requestTimeout | Wait time for response of a request. |
```

Comment:
Shouldn't these values be updated according to the changes in variable names?

Reply:
Missed it, done.